### PR TITLE
Complete CSV Generator & Postgres Seed Script

### PR DESCRIPTION
**Pull request by martinsung00**
**Size:** Large
**Related Trello Cards:** [card one](https://trello.com/c/o5FaIweM), [card two](https://trello.com/c/MB6NdyDD)
Change(s):

Revise Seeder Script (now an sql file)
Create Data Generator
Create Example in vault
Create Example in seed_data
Create Temporary SH bootstrap file
Update package.json
Update front-end slightly to successfully retrieve items from database
Update server routes to serve parsed response
Update tests to reflect changes
Prune unnecessary code

**Notes:**
Successfully generated 10 million records and seed database

### CODE REVIEW

File: seeder/postgres_seeder/data_generator/config.js [hunk 1]
```diff
@@ -20,10 +20,10 @@ Configurable options:
 */
 
 const config = {
-  product_range: 10,
+  product_range: 95239,
```
Comment (martinsung00) [inline seeder/postgres_seeder/data_generator/config.js:23]: **Annotation:** 95,239 products equate to 10+ million records as it branches into 7 different styles and each style branches off into 15 or 17 sizes depending on gender.

File: seeder/postgres_seeder/data_generator/seed_data/sample.csv [hunk 2]
```diff
@@ -0,0 +1,76 @@
+product_id, style_id, size, quantity
```
Comment (martinsung00) [inline seeder/postgres_seeder/data_generator/seed_data/sample.csv:1]: **Annotation:** This is just an example file for anyone else who gets this repository.

File: src/controller/logger.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import winston from "winston";
```
Comment (martinsung00) [inline src/controller/logger.js:1]: **Annotation:** Created a Winston logger specifically for controllers.

File: src/controller/plugin.js [hunk 4]
```diff
@@ -11,7 +14,7 @@ export default class Controller {
 
       return response;
     } catch (err) {
-      this.gateway.logger.log("error", "Error:", err);
```
Comment (martinsung00) [inline src/controller/plugin.js:14]: **Annotation:** Controllers should not be using gateway's logger, but it's own logger.

File: src/gateways/schema/schema.sql [hunk 5]
```diff
@@ -1,9 +1,10 @@
-CREATE DATABASE inventory
+DROP DATABASE IF EXISTS inventory;
+
+CREATE DATABASE inventory;
 
 CREATE TABLE IF NOT EXISTS products (
-  product_id varchar(250) NOT NULL,
-  style_id varchar(250) NOT NULL,
+  product_id varchar(50000) NOT NULL,
```
Comment (martinsung00) [inline src/gateways/schema/schema.sql:6]: **Annotation:** Testing limits.

File: src/server/routes.js [hunk 6]
```diff
@@ -47,7 +47,7 @@ app.get("/inventory/:productID/:styleID", function (req, res) {
   controller
     .read(pID, sID)
     .then(function (result) {
-      res.send(result).status(200);
+      res.send(result.rows).status(200);
```
Comment (martinsung00) [inline src/server/routes.js:50]: **Annotation:** Update server to serve array of information directly to reduce load on client.

File: seeder/postgres_seeder/data_generator/plugin.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    const quantity = Math.floor(Math.random() * Math.floor(100));
+
+    return quantity;
+  }
+
+  coinFlip() {
+    const result = Math.floor(Math.random() * 2);
+
+    return result;
+  }
```
Comment (TandyN) [inline seeder/postgres_seeder/data_generator/plugin.js]: Could technically return the results right away. At least I think its just as readable in 1 line.

**Review by TandyN — CHANGES_REQUESTED** [review]
Assuming that changing the API response to rows didn't break anything in the front end, I see no conflicts in functionality.

1 very minor suggestion was made in a file. Also, you've committed your bundle file unless that was intentional? if so, please explain. If not, please remove.

File: seeder/postgres_seeder/data_generator/plugin.js (hunk 7 above)
Comment (martinsung00) [reply seeder/postgres_seeder/data_generator/plugin.js]: Fixed it to return statement instead of variable.

Comment (martinsung00) [reply seeder/postgres_seeder/data_generator/plugin.js]: Fixed the gitignore not working properly with bundle.js; now it's fully ignored. The front-end works with the new changes in the server; made some slight edits beforehand.

**Review by TandyN — APPROVED** [review]
Requested changes have been applied. Please squash your commits.

**Comment by LesliePajuelo** [comment]
@martinsung00 hi! Martin you've had several large PRs! This is no longer acceptable. Come to office hours for help on how to manage your git workflow.

**Pull request merged**